packages/x-date-pickers-pro/src/MultiInputDateRangeField/MultiInputDateRangeField.tsx

```diff
@@ -13,17 +35,100 @@ export const MultiInputDateRangeField = React.forwardRef(function MultiInputDate
   TInputDate,
   TDate = TInputDate,
 >(inProps: MultiInputDateRangeFieldProps<TInputDate, TDate>, ref: React.Ref<HTMLInputElement>) {
-  const { startDate, endDate } = useMultiInputDateRangeField<
-    TInputDate,
-    TDate,
-    MultiInputDateRangeFieldProps<TInputDate, TDate>
-  >(inProps);
+  const themeProps = useThemeProps({
+    props: inProps,
+    name: 'MuiMultiInputDateRangeField',
+  });
+
+  const {
+    components,
+    componentsProps,
+    value,
+    defaultValue,
+    format,
+    onChange,
+    readOnly,
+    onError,
+    shouldDisableDate,
+    minDate,
+    maxDate,
+    disableFuture,
+    disablePast,
+    ...other
+  } = themeProps;
+
+  const ownerState = themeProps;
+
+  const Root = components?.Root ?? MultiInputDateRangeFieldRoot;
+  const rootProps = useSlotProps({
+    elementType: Root,
+    externalSlotProps: componentsProps?.root,
+    externalForwardedProps: other,
+    additionalProps: {
+      ref,
+    },
+    ownerState,
+  });
+
+  const Input = components?.Input ?? TextField;
+  const startInputProps = useSlotProps({
+    elementType: Input,
+    externalSlotProps: componentsProps?.input,
+    ownerState: { ...ownerState, position: 'start' },
+  });
+  const endInputProps = useSlotProps({
+    elementType: Input,
+    externalSlotProps: componentsProps?.input,
+    ownerState: { ...ownerState, position: 'end' },
+  });
+
+  const Separator = components?.Separator ?? MultiInputDateRangeFieldSeparator;
+  const separatorProps = useSlotProps({
+    elementType: Separator,
+    externalSlotProps: componentsProps?.separator,
+    ownerState,
+  });
+
+  const {
+    startDate: { onKeyDown: onStartInputKeyDown, ref: startInputRef, ...startDateProps },
+    endDate: { onKeyDown: onEndInputKeyDown, ref: endInputRef, ...endDateProps },
+  } = useMultiInputDateRangeField({
+    props: {
+      value,
+      defaultValue,
+      format,
+      onChange,
+      readOnly,
+      onError,
+      shouldDisableDate,
+      minDate,
+      maxDate,
+      disableFuture,
+      disablePast,
+    },
+    startInputRef: startInputProps.ref,
+    endInputRef: endInputProps.ref,
+  });
 
   return (
-    <Stack spacing={2} direction="row" alignItems="center" ref={ref}>
-      <TextField inputRef={startDate.inputRef} {...startDate.inputProps} />
-      <Typography>to</Typography>
-      <TextField inputRef={endDate.inputRef} {...endDate.inputProps} />
-    </Stack>
+    <Root {...rootProps}>
+      <Input
+        {...startDateProps}
+        inputProps={{
+          // ...startDateProps.inputProps,
+          ref: startInputRef,
+          onKeyDown: onStartInputKeyDown,
+        }}
+      />
```

Comment:
I don't get why the `startDateProps` is commented and `startInputProps` does not appears
```suggestion
      <Input
        {...startDateProps}
        {...startInputProps}
        inputProps={{
          ...startDateProps.inputProps,
          ...startInputProps.inputProps,
          ref: startInputRef,
          onKeyDown: onStartInputKeyDown,
        }}
      />
```

Reply:
I was not supporting correctly the props passed to each input
It should be better now.